mongo/integration/crud_prose_test.go

```diff
@@ -155,39 +156,54 @@ func TestAggregatePrimaryPreferredReadPreference(t *testing.T) {
 		MinServerVersion("4.1.0") // Consistent with tests in aggregate-out-readConcern.json
 
 	mt := mtest.New(t, mtOpts)
-	mt.Run("aggregate $out with non-primary read ppreference", func(mt *mtest.T) {
+	mt.Run("aggregate $out with non-primary read preference", func(mt *mtest.T) {
 		doc, err := bson.Marshal(bson.D{
 			{"_id", 1},
 			{"x", 11},
 		})
 		assert.Nil(mt, err, "Marshal error: %v", err)
-		_, err = mt.Coll.InsertOne(mtest.Background, doc)
-		assert.Nil(mt, err, "InsertOne error: %v", err)
-
-		mt.ClearEvents()
 		outputCollName := "aggregate-read-pref-primary-preferred-output"
-		outStage := bson.D{
-			{"$out", outputCollName},
+		testCases := []struct {
+			name     string
+			pipeline interface{}
+		}{
+			{
+				"pipeline",
+				mongo.Pipeline{bson.D{{"$out", outputCollName}}},
```

Comment:
Nice! This makes the test cases clearer. If it makes sense, can we add more cases for all ways a pipeline can be passed? IIUC that could also be a slice of BSON documents, and a `bson.A`, etc.

Reply:
done